websites/S/Sorryfield/presence.ts

```diff
@@ -0,0 +1,178 @@
+const presence = new Presence({
+	clientId: "1016312551958642698",
+});
+
+const enum Assets {
+	LOGO = "https://i.imgur.com/TxuMfTd.png",
+	CHART_SEARCH = "https://i.imgur.com/IyyYNPb.jpg",
+	CHART_HYPORANKED = "https://i.imgur.com/RDMG4bg.jpg",
+	CHART_EXPONENTIAL = "https://i.imgur.com/MAi9c8K.jpg",
+	CHART_HARD = "https://i.imgur.com/6NkL3hh.jpg",
+	CHART_INTERMEDIATE = "https://i.imgur.com/hRMaDwD.jpg",
+	CHART_EASY = "https://i.imgur.com/BLQqUN1.jpg",
+	CHART_NEW = "https://i.imgur.com/PpA5uhv.jpg",
+	CHART_HISTORY = "https://i.imgur.com/LxwfWhY.jpg",
+	CHART_POSSESSION = "https://i.imgur.com/C42Fbsi.jpg",
+	CHART_HOT = "https://i.imgur.com/xt6IZxO.jpg",
+	CHART_COLD = "https://i.imgur.com/E9xnBoS.jpg",
+	CHART_MULTIPLAYER = "https://i.imgur.com/fdGfvt2.jpg",
+}
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			details: "Sorryfield",
+			smallImageKey: Assets.LOGO,
+		},
+		{ pathname, href } = document.location;
+	if (
+		pathname === "/" ||
+		pathname === "/sorrygle" ||
+		pathname.startsWith("/song/")
+	) {
+		presenceData.largeImageKey = Assets.LOGO;
+		delete presenceData.smallImageKey;
+	}
+	if (pathname === "/") presenceData.details = "곡 선택 중";
+	if (pathname === "/sorrygle") presenceData.details = "쏘리글";
+	if (pathname.startsWith("/song/")) {
+		presenceData.details = "노래 듣는 중";
+		let songArtist = document
+			.querySelector(".song")
+			.children[2].innerHTML.split("&nbsp;")[0];
+		/* Why innerHTML?
+		 * Because removing another expression of artist.
+		 * Other expressions of artist name and song name are separated by '&nbsp;' in Sorryfield.
+		 * '.textContent' does not have '&nbsp;'.
+		 */
+		if (document.querySelector(".title").textContent.includes("fa-lock"))
+			songArtist = `🔒 ${songArtist}`; // Hidden song from the search.
+		presenceData.state = `${songArtist} - ${document
+			.querySelector(".title")
+			.textContent.trim()}`;
+		presenceData.buttons = [{ label: "Listen", url: href }];
```

Comment:
I know I partially requested this, but perhaps to be consistent, this should also be in Korean.